.buildkite/commands/set-up-git-for-private-repos.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash -eu
+
+if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
+  echo "This script must be sourced, not executed, because it exports GIT_SSH_COMMAND."
+  exit 1
+fi
+
+echo "--- :git: Change Git SSH key to fetch private dependencies"
```

Comment:
If I'm not mistaken, technically speaking this doesn't _change_ the SSH key (as in: _replacing_ it) but _adds_ it to the agent? (Meaning SSH will still try other keys in the agent if the first one(s) it tries fails?)

Reply:
It changes `GIT_SSH_COMMAND` so I think it's an accurate comment

Reply:
Ah, right 👍

I think I conflated the use of `-i` alone (which would add the identity but, if used alone, still let `ssh` try other identities registered in the agent or the config file) with when `-i` is used in association with `-o IdentitiesOnly=yes` (like here) to ask the `ssh-agent` not to offer any keys, and only use the keys provided by `-i` on the command line, or by the config files.

_To be even stricter we should also add `-F /dev/null` to also tell ssh not to use any keys configured in the config file either, to be sure that the only key it will proposed is the one provided via `-i` and not try any other key boforehand; but in the case of CI and how Buildkite works wrt to exposing keys to SSH, I don't think any Identity would be configured in the `~/.ssh/config` of our agents and risk being used and tried before the bot's private key provided here… so ultimately `-F /dev/null` would probably unnecessary in this context._